terminal/TerminalClient.cpp

```diff
@@ -165,13 +165,36 @@ int main(int argc, char** argv) {
       auto j = split(FLAGS_portforward, ',');
       for (auto& pair : j) {
         vector<string> sourceDestination = split(pair, ':');
-        // TODO: Handle bad input
-        int sourcePort = stoi(sourceDestination[0]);
-        int destinationPort = stoi(sourceDestination[1]);
-
-        portForwardRouter.addListener(
-            shared_ptr<PortForwardClientListener>(new PortForwardClientListener(
-                socketHandler, sourcePort, destinationPort)));
+	if (sourceDestination[0].find('-') != string::npos
+	    && sourceDestination[1].find('-') != string::npos) {
+	  vector<string> sourcePortRange = split(sourceDestination[0], '-');
+	  int sourcePortStart = stoi(sourcePortRange[0]);
+	  int sourcePortEnd = stoi(sourcePortRange[1]);
+
+	  vector<string> destinationPortRange = split(sourceDestination[1], '-');
+	  int destinationPortStart = stoi(destinationPortRange[0]);
+	  int destinationPortEnd = stoi(destinationPortRange[1]);
+
+	  if (sourcePortEnd - sourcePortStart != destinationPortEnd - destinationPortStart) {
+	    cout << "source/destination port range don't match" << endl;
+	    exit(1);
+	  } else {
+	    int portRangeLength = sourcePortEnd - sourcePortStart + 1;
+	    for (int i=0; i < portRangeLength; ++i) {
+	      portForwardRouter.addListener(
+		  shared_ptr<PortForwardClientListener>(new PortForwardClientListener(
+		      socketHandler, sourcePortStart+i, destinationPortStart+i)));
+	    }
+	  }
+	} else {
+          // TODO: Handle bad input
```

Comment:
Can you also handle bad input as part of this PR?

All you should need to do is add catches for std::invalid_argument and std::out_of_range to the catch clause below.  Printing to cout and exit(1) should be sufficient.

Thanks!

Reply:
Sure!

Reply:
Actually if you want to only add 1 more catch, you can use std::logic_error which includes both of the ones I mentioned above.